wrapper/src/main/java/software/amazon/jdbc/util/ConnectionUrlBuilder.java

```diff
@@ -22,73 +22,78 @@
 import java.sql.SQLException;
 import java.util.Enumeration;
 import java.util.Properties;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
 import software.amazon.jdbc.HostSpec;
 import software.amazon.jdbc.PropertyDefinition;
 
 public class ConnectionUrlBuilder {
 
+  // Builds a connection URL of the generic format: "protocol//[hosts][:port]/[database]"
+  public static String buildUrl(
+      final @NonNull String jdbcProtocol,
+      final @NonNull String serverName,
+      final int port,
+      final @Nullable String databaseName) throws SQLException {
+
+    if (StringUtils.isNullOrEmpty(jdbcProtocol) || StringUtils.isNullOrEmpty(serverName)) {
+      throw new SQLException(Messages.get("ConnectionUrlBuilder.missingJdbcProtocol"));
```

Comment:
The message says either jdbc protocol or server name is missing. `ConnectionUrlBuilder.missingJdbcProtocol` might be not the best message ID since it doesn't mention server name.